libs/agno/agno/knowledge/pdf_url.py

```diff
@@ -1,40 +1,65 @@
-from typing import AsyncIterator, Iterator, List, Union
+from typing import Any, AsyncIterator, Dict, Iterator, List, Optional, Union
 
 from agno.document import Document
 from agno.document.reader.pdf_reader import PDFUrlImageReader, PDFUrlReader
 from agno.knowledge.agent import AgentKnowledge
-from agno.utils.log import logger
+from agno.utils.log import log_info, logger
 
 
 class PDFUrlKnowledgeBase(AgentKnowledge):
-    urls: List[str] = []
+    urls: Optional[Union[List[str], List[Dict[str, Dict[str, Any]]]]] = None
+    formats: List[str] = [".pdf"]
     reader: Union[PDFUrlReader, PDFUrlImageReader] = PDFUrlReader()
 
     @property
     def document_lists(self) -> Iterator[List[Document]]:
-        """Iterate over PDF urls and yield lists of documents.
-        Each object yielded by the iterator is a list of documents.
+        """Iterate over PDF urls and yield lists of documents."""
+        if not self.urls:
+            raise ValueError("URLs are not set")
 
-        Returns:
-            Iterator[List[Document]]: Iterator yielding list of documents
-        """
-
-        for url in self.urls:
-            if url.endswith(".pdf"):
-                yield self.reader.read(url=url)
+        for item in self.urls:
+            if isinstance(item, dict):
+                # Handle URL with metadata
+                for url, config in item.items():
+                    if self._is_valid_url(url):
+                        documents = self.reader.read(url=url)
+                        # Add metadata to documents if provided
+                        if config.get("metadata"):
+                            for doc in documents:
+                                log_info(f"Adding metadata {config.get('metadata')} to document from URL: {url}")
+                                doc.meta_data.update(config["metadata"])
+                        yield documents
             else:
-                logger.error(f"Unsupported URL: {url}")
+                # Handle simple URL
+                if self._is_valid_url(item):
+                    yield self.reader.read(url=item)
+
+    def _is_valid_url(self, url: str) -> bool:
+        """Helper to check if URL is valid."""
+        if not url.endswith(".pdf"):
+            logger.error(f"Unsupported URL: {url}")
+            return False
+        return True
```

Comment:
As this is a type validation, it would be more user friendly to be clear about it. Like:
```python
def _is_valid_file_type(...):
    [...]
    logger.error(f"Unsupported file type in URL: {url}")
```